api/v1beta1/hostedcluster_types.go

```diff
@@ -444,38 +480,49 @@ type RoutePublishingStrategy struct {
 }
 
 // DNSSpec specifies the DNS configuration in the cluster.
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.publicZoneID) || has(self.publicZoneID)", message="publicZoneID is required once set"
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.privateZoneID) || has(self.privateZoneID)", message="privateZoneID is required once set"
 type DNSSpec struct {
 	// BaseDomain is the base domain of the cluster.
 	//
 	// +immutable
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="baseDomain is immutable"
 	BaseDomain string `json:"baseDomain"`
 
 	// PublicZoneID is the Hosted Zone ID where all the DNS records that are
 	// publicly accessible to the internet exist.
 	//
 	// +optional
 	// +immutable
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="publicZoneID is immutable"
```

Comment:
are publicZoneID and PrivateZoneID actually optional? cc @sjenning

Reply:
Looking at the source code, I think you are right.  Would be nice to hear from Seth  to confirm though.

Reply:
Commented in slack, but I think this is only required if platform is AWS.  It get piped into the DNS global config in the guest cluster.